### PR TITLE
REF use pyproject.toml

### PR DESCRIPTION
**Pull request by beckermr**
This PR converts the project to use only pyproject.toml.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,8 +1,5 @@
-oldest-supported-numpy
```
Comment (erykoff) [inline requirements.txt:1]: I guess I don't need this magic package for a python-only thing, but we should declare numpy, right?

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,3 @@
+from setuptools import setup
+
+setup()
```
Comment (erykoff) [inline setup.py:3]: Do we really need a `setup.py` file?

File: pyproject.toml [hunk 3]
```diff
@@ -2,9 +2,56 @@
 requires = [
     "setuptools>=46",
     "wheel",
-    "setuptools_scm",
-    "setuptools_scm_git_archive"
+    "setuptools_scm>=7",
```
Comment (erykoff) [inline pyproject.toml:5]: Does this now cover the git stuff?

File: pyproject.toml [hunk 4]
```diff
@@ -2,9 +2,56 @@
...
+    "tomli>=1.0.0; python_version < '3.11'",
```
Comment (erykoff) [inline pyproject.toml:6]: What is this about?

File: pyproject.toml (hunk 3 above)
Comment (beckermr) [reply pyproject.toml:5]: yes it does!

File: setup.py (hunk 2 above)
Comment (beckermr) [reply setup.py:3]: This is to help folks with backwards compatibility. We don't have to keep it, but it is a nice courtesy.

File: .github/workflows/python-package-pr-pip.yaml [hunk 5]
```diff
@@ -19,14 +19,16 @@ jobs:
...
           python-version: "3.10"
           cache: "pip"
-          cache-dependency-path: "setup.cfg"
+          cache-dependency-path: "requirements.txt"
 
       - name: Build and install
         run: |
           python -m pip install --upgrade pip setuptools
           python -m pip install pytest
           if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
```
Comment (beckermr) [inline .github/workflows/python-package-pr-pip.yaml:28]:
```suggestion
```

File: requirements.txt (hunk 1 above)
Comment (beckermr) [reply requirements.txt:1]: yep! fixed!

File: pyproject.toml (hunk 4 above)
Comment (beckermr) [reply pyproject.toml:6]: IDK actually. I have been cargo culting this around for a while now.

**Review by erykoff — APPROVED** [review]
This looks fine, and I trust you that this is the Way.  But one thing that I have been and still am confused about is the requirements that are listed in both the `pyproject.toml` and the requirements files.  I thought that if this is done right these sorts of duplications were no longer needed (not that I've actually figured that out).

**Comment by beckermr** [comment]
Which duplicates?

**Comment by erykoff** [comment]
Maybe they are separate, I was just wondering why (e.g.) tomli and setuptools_scm are listed in build reqs and dev req file:
```
requires = [
    "setuptools>=46",
    "wheel",
    "setuptools_scm>=7",
    "tomli>=1.0.0; python_version < '3.11'",
]
```

**Comment by beckermr** [comment]
Ah right. This duplication is not possible to fix. 

https://stackoverflow.com/questions/75131838/specifying-build-requirements-as-a-file-in-a-setuptools-pyproject-toml

**Pull request merged**